x-pack/plugins/security_solution/server/lib/detection_engine/rule_management/logic/import/import_rules_utils.ts

```diff
@@ -104,18 +106,41 @@ export const importRules = async ({
               });
 
               if (rule == null) {
-                await createRules({
+                const createdRule = await createRules({
                   rulesClient,
                   params: {
                     ...parsedRule,
                     exceptions_list: [...exceptions],
                   },
                   allowMissingConnectorSecrets,
                 });
-                resolve({
-                  rule_id: parsedRule.rule_id,
-                  status_code: 200,
+                // find rules with the same rule_id, in case of concurrent imports
+                const rules = await findRules({
+                  rulesClient,
+                  filter: `alert.attributes.params.ruleId: "${parsedRule.rule_id}"`,
+                  page: 1,
+                  fields: undefined,
+                  perPage: undefined,
+                  sortField: 'created_at',
+                  sortOrder: 'asc',
                 });
+                // OK if it is a unique rule or the earlier rule created
+                if (rules.total === 1 || createdRule.id === rules.data[0].id) {
+                  resolve({
+                    rule_id: parsedRule.rule_id,
+                    status_code: 200,
+                  });
+                  // else delete the duplicate rule and return status 409
+                } else {
+                  deleteRules({ ruleId: createdRule.id, rulesClient });
```

Comment:
`deleteRules()` may fail which will result to multiple rules with the same ruleId.